### ISSUE TITLE
Broken settings validation on `nCycles`

### ISSUE THREAD

**Issue by keckler** [terrapower/armi#635]
When the settings validation takes place, ARMI complains if you enter `nCycles=0` and no other cycle information:
```
$ armi run anl-afci-177.yaml

       +===================================================+
       |            _      ____     __  __    ___          |
       |           / \    |  _ \   |  \/  |  |_ _|         |
       |          / _ \   | |_) |  | |\/| |   | |          |
       |         / ___ \  |  _ <   | |  | |   | |          |
       |        /_/   \_\ |_| \_\  |_|  |_|  |___|         |
       |        Advanced  Reactor  Modeling Interface      |
       |                                                   |
       |                    version 0.2.3                  |
       |                                                   |
       +===================================================+
[prmt] INSPECTOR: Cannot run 0 cycles. Set burnSteps to 0 to activate a single time-independent case.
[prmt] Set 1 cycle and 0 burnSteps for single time-independent case? (YES, NO, CANCEL, Y, N):
```

If you tell ARMI to go ahead and make the suggested change, it continues without actually making the change.

The lines at fault are here:

https://github.com/terrapower/armi/blob/658900ff8d344676e6e924476ba499b3a5ebb6ea/armi/operators/settingsValidation.py#L500-L509

In particular, the `_correctCycles` function's use of `self.cs.modified(....)` doesn't work for some reason. I don't really understand the reason for this deficiency, but apparently it should instead be using `_assignCS`:

https://github.com/terrapower/armi/blob/658900ff8d344676e6e924476ba499b3a5ebb6ea/armi/operators/settingsValidation.py#L261-L265

**Comment by keckler** [comment]
I will take care of fixing this in conjunction with some other work I'm doing.

**Comment by keckler** [comment]
Handled in https://github.com/keckler/armi/commit/c4480081ce40c5855b1170f424e4b98efa32303b

**Comment by keckler** [comment]
Closed with the closure of #637
